### PR TITLE
Refresh port info update mechanism and other general upgrades

### PR DESCRIPTION
**Pull request by arjunsalyan**

### CODE REVIEW

File: app/port/models.py [hunk 1]
```diff
@@ -48,151 +48,33 @@ def get_absolute_url(self):
...
-                        categories.add(Category(name=category))
-                except KeyError:
-                    pass
-            batch = list(categories)
-            Category.objects.bulk_create(batch)
-            return
-
-        port_id_map = {}
+    def add_or_update(cls, data):
+        from category.models import Category
```
Comment (rajgoesout) [inline app/port/models.py]: Is there any performance gain by importing inside the function? If not, then I would keep all the imports at the top https://www.python.org/dev/peps/pep-0008/

**Review by rajgoesout — COMMENTED** [review]
LGTM

File: app/port/models.py (hunk 1 above)
Comment (arjunsalyan) [reply app/port/models.py]: I did this to prevent cyclic imports. However, I will try if I can get them at the top, if there are no other conflicts.

Comment (rajgoesout) [reply app/port/models.py]: You could use `import foo as bar` if there is a name clash/circular import.

**PS.** this isn't super important

Comment (arjunsalyan) [reply app/port/models.py]: Thank you, I had fixed the cyclic imports earlier itself but forgot to change the import statements. This has been fixed.

**Pull request merged**